code/modules/power/furnace.dm

```diff
@@ -310,3 +295,51 @@ TYPEINFO(/obj/machinery/power/furnace)
 			qdel(W)
 
 		return 1
+
+/datum/action/bar/icon/stuff_mob_into_furnace
+	id = "stuff_into_furnace"
+	duration = 5 SECONDS
+	interrupt_flags = INTERRUPT_ACT | INTERRUPT_STUNNED | INTERRUPT_MOVE | INTERRUPT_ATTACKED
+	var/obj/machinery/power/furnace/furnace = null
+	var/mob/victim = null
+	var/mob/user = null
+
+	New(mob/U, obj/machinery/power/furnace/F, mob/V)
+		..()
+		src.user = U
+		src.furnace = F
+		src.icon = F.icon
```

Comment:
The furnace icon looks really weird here, probably better to use a generic icon like the pocket searching action bar.